File: peak_element_array.cpp
// Time Complexity :O(logn)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this :No


// Your code here along with comments explaining your approach

class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int left=0,right=nums.size()-1;
        while(left<right)
        {
            int mid=left+(right-left)/2;
            if(nums[mid]<nums[mid+1])
                left=mid+1;
            else
                right=mid;
        }
        return left;
    }
};

File: Peak_Element_Array.java
//TC: O(logn)
//SC: O(1)


class Solution {
    public int findPeakElement(int[] arr) {
        int low = 0;
        int high = arr.length -1;
        while (low < high) {
            int mid = low + (high - low)/2;
            if(arr[mid]<arr[mid+1]){
                low = mid+1;
            } else {
                high = mid;
            }
        }
        return low;
    }
}
